akka-actor-typed/src/main/scala/akka/actor/typed/internal/DelayedStart.scala

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.actor.typed.internal
+
+import akka.actor.typed.{ ActorContext, Behavior, Signal, scaladsl }
+import akka.annotation.InternalApi
+
+/**
+ * INTERNAL API
+ *
+ * A behavior that wraps another behavior, stashing all incoming messages and signals until
+ * it is sent the `Start` message, it then unstashes all messages and signals and switches to the delayedBehavior
+ */
+@InternalApi
+private[akka] object DelayedStart {
+
+  case object Start
+
+  // note that we cannot use StashBuffer here as we need to buffer signals as well as messages
+  def delayStart[T](delayedBehavior: Behavior[T], buffer: List[Any] = Nil): Behavior[Any] =
+    scaladsl.Behaviors.immutable[Any] { (sctx, msg) ⇒
+      // delay actual initialization until the actor system is started as the actor may touch
+      // other parts of the actor system which would not be ready unless delayed
+      val ctx = sctx.asInstanceOf[ActorContext[T]]
+      msg match {
+        case Start ⇒
+          val endState = buffer.reverseIterator.foldLeft(Behavior.undefer(delayedBehavior, ctx)) { (behavior, buffered) ⇒
```

Comment:
If you end up not using the MutableStashBuffer you should at least use the new `Behavior.interpretMessages(iter)`

Reply:
Got it!

Reply:
Oh, but wait, I have messages mixed with signals and need to interpret them in the right order,  `interpretMessages` sounds like it only deals with messages?

Reply:
it actual handles both

and since signals typically are triggered by system messages there would not even be a guarantee of ordering between signals and messages (system messages are processed first via a separate mailbox)

Reply:
Ok, I realise I was actually thinking about the wrong thing as well, because what I'm after is that if the behavior stops it won't process more messages, which doesn't really have anything to do with signals/PostStop but with message + behavior.